### PR TITLE
Implemented input mask feature

### PR DESCRIPTION
**Pull request by gergo85**
Website of plugin: https://github.com/igorescobar/jQuery-Mask-Plugin

### CODE REVIEW

**Comment by w20k** [comment]
Could you also provide a usage documentation for OctoberCMS community? :)
Would be great to have one, maybe with tips & tricks, or common mask patterns - https://github.com/octobercms/docs.

Let's see, what @LukeTowers  would say about this feature ;)

**Comment by gergo85** [comment]
I made the documentation :)

File: modules/backend/widgets/Form.php [hunk 1]
```diff
@@ -160,6 +160,7 @@ public function bindToController()
     protected function loadAssets()
     {
         $this->addJs('js/october.form.js', 'core');
+        $this->addJs('/modules/backend/assets/vendor/mask/jquery.mask.min.js', 'core');
```
Comment (LukeTowers) [inline modules/backend/widgets/Form.php]: Does this support subfolder installations?

File: modules/backend/classes/FormField.php [hunk 2]
```diff
@@ -449,11 +461,27 @@ protected function filterAttributes($attributes, $position = 'field')
         if ($position == 'field' && $this->readOnly) {
             $attributes = $attributes + ['readonly' => 'readonly'];
 
+            // https://stackoverflow.com/questions/155291/can-html-checkboxes-be-set-to-readonly
             if ($this->type == 'checkbox' || $this->type == 'switch') {
                 $attributes = $attributes + ['onclick' => 'return false;'];
             }
         }
 
+        if ($position == 'field' && $this->mask != '' && $this->type == 'text') {
```
Comment (LukeTowers) [inline modules/backend/classes/FormField.php]: Don't use `==` please, that's just asking for potential headaches

Comment (w20k) [reply modules/backend/classes/FormField.php]: Most probably @LukeTowers was saying to change == -> === :) 'empty' check also good one 👍

Comment (gergo85) [reply modules/backend/classes/FormField.php]: Thank you for answer. I modified the code.

Comment (LukeTowers) [reply modules/backend/classes/FormField.php]: @w20k is correct @gergo85; thanks to both of you.

**Comment by gergo85** [comment]
Is it correct?

File: modules/backend/widgets/Form.php (hunk 1 above)
Comment (LukeTowers) [reply modules/backend/widgets/Form.php]: @gergo85 did you test on a install of October where it lives in /cms? I.e. `example.com/cms/modules...`

Comment (gergo85) [reply modules/backend/widgets/Form.php]: @LukeTowers I installed a new October into subfolder. I tested this feature and it works perfectly. Could I send an account for demo webpage?

**Comment by LukeTowers** [comment]
@gergo85 I'm not 100% sure about the way of specifying the options. Do you have a list of all options that are available in that plugin?

**Comment by gergo85** [comment]
Options list and documentation: https://github.com/octobercms/docs/pull/319/commits/2ff48f5d8b04418c86644e79a452f54540cda2ac

**Comment by LukeTowers** [comment]
Is that all of the options that jQuery plugin offers though?

**Comment by w20k** [comment]
Three of those (reverse, clearIfNotMatch and selectOnFocus) and 'placeholder', which is also a nice feature ;)

**Comment by gergo85** [comment]
Documentation of plugin: https://igorescobar.github.io/jQuery-Mask-Plugin/docs.html

I implemented those options, which have the data-mask attribute.

**Comment by gergo85** [comment]
@LukeTowers Is it right?

**Comment by LukeTowers** [comment]
@gergo85 sorry for the delay on this, I need to review this against the docs thoroughly to make sure that it's setup in the best way possible.

**Comment by gergo85** [comment]
@LukeTowers Alright, I look forward to your feedback. Thank you for your effort!

**Comment by gergo85** [comment]
@LukeTowers Do you have free time to thinking about this feature?

**Comment by gergo85** [comment]
I made this feature as a plugin. It part of [Backend Elite](https://octobercms.com/plugin/indikator-backendelite) plugin.

**Pull request closed without merging**